src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseSigner.cs

```diff
@@ -5,23 +5,60 @@
+    /// <seealso cref="CoseMultiSignMessage.AddSignatureForEmbedded(CoseSigner, byte[])"/>
     public sealed class CoseSigner
     {
         internal readonly KeyType _keyType;
         internal readonly int? _algHeaderValueToSlip;
         internal CoseHeaderMap? _protectedHeaders;
         internal CoseHeaderMap? _unprotectedHeaders;
+
+        /// <summary>
+        /// Gets the cryptographic key of this instance.
+        /// </summary>
         public AsymmetricAlgorithm Key { get; }
+
+        /// <summary>
+        /// Gets the hash algorithm of this instance.
+        /// </summary>
         public HashAlgorithmName HashAlgorithm { get; }
+
+        /// <summary>
+        /// Gets the signature padding of this instance.
+        /// </summary>
         public RSASignaturePadding? RSASignaturePadding { get; }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CoseSigner"/> class.
+        /// </summary>
+        /// <param name="key">The private key used to sign the content.</param>
+        /// <param name="hashAlgorithm">The hash algorithm used to hash the content.</param>
+        /// <param name="protectedHeaders">The protected header parameters to append to the message.</param>
+        /// <param name="unprotectedHeaders">The unprotected header parameters to append to the message.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="key"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="key"/> is <see cref="RSA"/>, use <see cref="CoseSigner(RSA, RSASignaturePadding, HashAlgorithmName, CoseHeaderMap?, CoseHeaderMap?)"/> to specify a signature padding.
+        /// -or-
+        /// <paramref name="key"/> is of an unsupported type.
+        /// -or-
+        /// <see cref="CoseHeaderLabel.Algorithm"/> was incorrect based on the <paramref name="key"/> and <paramref name="hashAlgorithm"/>.
+        /// -or-
+        /// <see cref="CoseHeaderLabel.Algorithm"/> was not contained in <paramref name="protectedHeaders"/>.</exception>
+        /// <remarks>
+        /// For sign operations in <see cref="CoseSign1Message"/>, <paramref name="protectedHeaders"/> and <paramref name="unprotectedHeaders"/> are used as the buckets of the content (and only) layer.
+        /// For sign operations in <see cref="CoseMultiSignMessage"/>, <paramref name="protectedHeaders"/> and <paramref name="unprotectedHeaders"/> are used as the buckets of the signature layer.
+        /// </remarks>
         public CoseSigner(AsymmetricAlgorithm key, HashAlgorithmName hashAlgorithm, CoseHeaderMap? protectedHeaders = null, CoseHeaderMap? unprotectedHeaders = null)
         {
             if (key is null)
                 throw new ArgumentNullException(nameof(key));
 
             if (key is RSA)
-                throw new CryptographicException(SR.CoseSignerRSAKeyNeedsPadding);
+                throw new ArgumentException(SR.CoseSignerRSAKeyNeedsPadding);
```

Comment:
Good change.

Reply:
Except, it needs to say which argument.

```suggestion
                throw new ArgumentException(SR.CoseSignerRSAKeyNeedsPadding, nameof(key));
```